tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-literals.bal

```diff
@@ -363,3 +363,36 @@ function testXMLReturnUnion() returns error? {
     test:assertTrue(xmlSequence is xml);
     test:assertEquals((check xmlSequence).toString(), "<grand_total>2</grand_total>");
 }
+
+type T xml;
+
+function testXMLInterpolationExprWithUserDefinedType() {
+    T x1 = xml `<foo></foo>`;
+    xml x2 = xml `<doc>${x1}</doc>`;
+    test:assertEquals(x2.toString(), "<doc><foo></foo></doc>");
+}
+
+function testQueryInXMLTemplateExpr() {
+    int[] a = [1, 2, 3];
+
+    var x1 = xml `<doc>${from int i in a select xml `<num>${i}</num>`}</doc>`;
+    xml x2 = xml `<doc>${from int i in a select xml `<num>${i}</num>`}</doc>`;
+    xml:Element x3 = xml `<doc>${from int i in a select xml `<num>${i}</num>`}</doc>`;
```

Comment:
Shall we add a few negative tests that throw incompatible type errors that show the type resolved from the query?